### ISSUE TITLE
Owncloud stops working

### ISSUE THREAD

**Issue by wesdonohue** [owncloud/core#14175]
I upgraded to 8 yesterday and every few hours since upgrading the entire owncloud site becomes inaccessible. The apache service is running and I don't see any errors in the apache log or the system logs. This is running on ubuntu 14.0. This was running with no issues on 7. I would add more information but since there isn't really anything relevant showing up in any logs I apologize that the information is limited. I'm looking through everything trying to find something I can share. Let me know if there is anything that would be helpful to look at and I will post it!

When I try to access the site it just sits like the page is loading but never loads. It will sit there like that until I close the browser. No error or connection timeout.

**Comment by LukasReschke** [comment]
Thanks a lot for reporting issues back to us!

We need some more information to properly support you with your issue!

Can I ask you to follow our guidelines for submitting bugs as described here: https://github.com/owncloud/core/blob/master/CONTRIBUTING.md

Thanks a lot!

**Comment by simsala** [comment]
There seem to be quite a bunch of us with the same problem.
Seems related to Ubuntu 14.04 with PHP 5.5.9.

See here:
http://forum.owncloud.org/viewtopic.php?f=31&t=26331

**Comment by RealRancor** [comment]
Hi,

there are some reports that this could be related to php5-acpu on those PHP 5.5.9 installations.

**Comment by LukasReschke** [comment]
From the Forum this looks like https://github.com/owncloud/core/issues/9885 which has been fixed with https://github.com/owncloud/core/pull/11056. Are you by any chance using APC / Zend OpCache?

Can you guys please test the changes in https://github.com/owncloud/core/pull/11056?

@karlitschek @DeepDiver1975 Possible serious backport candidate if this gets confirmed.

**Comment by LukasReschke** [comment]
http://forum.owncloud.org/viewtopic.php?f=31&t=26331&p=79102&hilit=zend_mm_heap#p79102  looks pretty much like it.

**Comment by LukasReschke** [comment]
Though there could as well be as usual multiple bugs into one single topic *sigh* – let's see what the persons in the forum report back – I asked in http://forum.owncloud.org/viewtopic.php?p=79439#p79439

**Comment by wesdonohue** [comment]
Awesome thanks guys! That looks like the same problem. I'll try downgrading php.

**Comment by AdamWill** [comment]
wes: you could try applying the change from https://github.com/owncloud/core/pull/11056 to your OC install instead - it's easy to manually hack in (or get the patch from github and apply) and it'd be useful to know if it helps.

**Comment by apastuszak** [comment]
Applied #11056 manually and it did not resolve my issue.

**Comment by enoch85** [comment]
Same issue here on one of my servers. 

```
Fri Feb 13 01:35:07.593106 2015] [:error] [pid 6986] [client 178.174.137.236:62056] PHP Warning:  apc_store(): GC cache entry 'ocdd06e12069/AutoloaderOC\\DB\\Adapter' was on gc-list for 3634 seconds in /var/www/owncloud/lib/private/memcache/apc.php on line 21
[Fri Feb 13 01:35:07.593127 2015] [:error] [pid 6986] [client 178.174.137.236:62056] PHP Warning:  apc_store(): GC cache entry 'ocdd06e12069/AutoloaderOCP\\IDBConnection' was on gc-list for 3634 seconds in /var/www/owncloud/lib/private/memcache/apc.php on line 21
[Fri Feb 13 01:35:07.593167 2015] [:error] [pid 6986] [client 178.174.137.236:62056] PHP Warning:  apc_store(): GC cache entry 'ocdd06e12069/AutoloaderOC\\DB\\Connection' was on gc-list for 3634 seconds in /var/www/owncloud/lib/private/memcache/apc.php on line 21
[Fri Feb 13 01:35:07.593186 2015] [:error] [pid 6986] [client 178.174.137.236:62056] PHP Warning:  apc_store(): GC cache entry 'ocdd06e12069/AutoloaderOC\\Session\\Memory' was on gc-list for 3634 seconds in /var/www/owncloud/lib/private/memcache/apc.php on line 21
[Fri Feb 13 01:35:07.593205 2015] [:error] [pid 6986] [client 178.174.137.236:62056] PHP Warning:  apc_store(): GC cache entry 'ocdd06e12069/AutoloaderOC\\Hooks\\PublicEmitter' was on gc-list for 3634 seconds in /var/www/owncloud/lib/private/memcache/apc.php on line 21
[Fri Feb 13 01:35:07.593223 2015] [:error] [pid 6986] [client 178.174.137.236:62056] PHP Warning:  apc_store(): GC cache entry 'ocdd06e12069/AutoloaderOC\\AllConfig' was on gc-list for 3634 seconds in /var/www/owncloud/lib/private/memcache/apc.php on line 21
[Fri Feb 13 01:35:07.593240 2015] [:error] [pid 6986] [client 178.174.137.236:62056] PHP Warning:  apc_store(): GC cache entry 'ocdd06e12069/AutoloaderOC\\Files\\Stream\\Close' was on gc-list for 3634 seconds in /var/www/owncloud/lib/private/memcache/apc.php on line 21
[Fri Feb 13 01:35:07.593258 2015] [:error] [pid 6986] [client 178.174.137.236:62056] PHP Warning:  apc_store(): GC cache entry 'ocdd06e12069/AutoloaderOC\\Files\\Stream\\Dir' was on gc-list for 3634 seconds in /var/www/owncloud/lib/private/memcache/apc.php on line 21
[Fri Feb 13 01:35:07.593276 2015] [:error] [pid 6986] [client 178.174.137.236:62056] PHP Warning:  apc_store(): GC cache entry 'ocdd06e12069/AutoloaderOC\\Log' was on gc-list for 3634 seconds in /var/www/owncloud/lib/private/memcache/apc.php on line 21
```

Then it crashes. Seems like it anyway.

**Comment by simsala** [comment]
This apc thing is not related to the Ubuntu 14.04 / PHP 5.5.9 issue.
My server hangs every now and then without apc. Besides that as far as I know, apc does not work with PHP 5.5.
I had opcache enabled on my system. Disabling it did not resolve the issue however.

**Comment by simsala** [comment]
What we are experiencing is:

[Thu Feb 12 00:29:37.506892 2015] [mpm_prefork:error] [pid 24568] AH00161: server reached MaxRequestWorkers setting, consider raising the MaxRequestWorkers setting

**Comment by LukasReschke** [comment]
Can you please all open new issues? We're mixing up different stuff in here and this won't help anybody. THX ;-)

**Comment by BoBeR182** [comment]
Hey everyone,

I was experiencing the same issue on a new installation, and after some digging around I have successfully concluded that the issue was the cache that was being used.

FYI im on ubuntu 14.04 server.

after running apt-cache depends owncloud i get the following list

Depends: owncloud-server
Depends: owncloud-config-apache
Depends: owncloud-3rdparty
Depends: owncloud-app-activity
Depends: owncloud-app-files
Depends: owncloud-app-files-encryption
Depends: owncloud-app-files-pdfviewer
Depends: owncloud-app-files-trashbin
Depends: owncloud-app-firstrunwizard
Depends: owncloud-app-templateeditor
Depends: owncloud-app-user-ldap
Depends: owncloud-app-external
Depends: owncloud-app-files-external
Depends: owncloud-app-files-sharing
Depends: owncloud-app-files-versions
Depends: owncloud-app-gallery
Depends: owncloud-app-updater
Depends: owncloud-app-user-webdavauth
Depends: owncloud-app-files-locking
Depends: owncloud-app-files-texteditor
Depends: owncloud-app-files-videoviewer
Depends: owncloud-app-provisioning-api
Depends: owncloud-app-user-external
|Suggests: clamav
Suggests: clamav-daemon
Suggests: smbclient
Suggests: libreoffice-writer
Recommends: mysql-server
Recommends: curl
curl:i386
|Recommends: exim4
Recommends: <mail-transport-agent>
citadel-mta
courier-mta
esmtp-run
exim4-daemon-light
lsb-invalid-mta
masqmail
msmtp-mta
nullmailer
opensmtpd
qmail-run
sendmail-bin
ssmtp
xmail
dma:i386
dma
exim4-daemon-heavy
postfix
|Recommends: php5-apcu
Recommends: <php5-apc>
Conflicts: <owncloud-fhs>
Conflicts: <owncloud-fhs:i386>

As you can see that php5-apc conflicts with the owncloud fhs so php5-apcu is installed instead. This was confirmed by running apt-cache policy on both php5-apc and php5-apcu.

Seeing as that is a recommended package I proceeded to purge it from the system. Owncloud functionality was not effected at all. My OC has been up for over 12 hours now.

Wanted to share this with you all as I hope it will help alot of you as it helped me.

**Comment by Akhenaton** [comment]
Hi,
[ownCloud 8.0.0.7]

Just to say that i have NOT this problem with PHP 5.6.5-1 / Zend Engine v2.6.0 / Zend OPcache v7.0.4-dev

**Comment by wesdonohue** [comment]
saw the same conflict with php5-apc. I've removed it, now waiting to see if that fixes the problem. Tried the other solutions yesterday with no change. We'll see if that is the fix for me as well.

**Comment by apastuszak** [comment]
So, to be clear, you are purging php5-apc or php5-apcu from the system?

**Comment by enoch85** [comment]
I have several conflicts, so what's your suggestion @BoBeR182 ?

```
root@fsgo:~# apt-cache depends owncloud
owncloud
  Depends: owncloud-server
  Depends: owncloud-config-apache
  Depends: owncloud-3rdparty
  Depends: owncloud-app-activity
  Depends: owncloud-app-files
  Depends: owncloud-app-files-encryption
  Depends: owncloud-app-files-pdfviewer
  Depends: owncloud-app-files-trashbin
  Depends: owncloud-app-firstrunwizard
  Depends: owncloud-app-templateeditor
  Depends: owncloud-app-user-ldap
  Depends: owncloud-app-external
  Depends: owncloud-app-files-external
  Depends: owncloud-app-files-sharing
  Depends: owncloud-app-files-versions
  Depends: owncloud-app-gallery
  Depends: owncloud-app-updater
  Depends: owncloud-app-user-webdavauth
  Depends: owncloud-app-files-locking
  Depends: owncloud-app-files-texteditor
  Depends: owncloud-app-files-videoviewer
  Depends: owncloud-app-provisioning-api
  Depends: owncloud-app-user-external
 |Suggests: clamav
  Suggests: clamav-daemon
  Suggests: smbclient
  Suggests: libreoffice-writer
  Recommends: mysql-server
  Recommends: curl
    curl:i386
 |Recommends: exim4
  Recommends: <mail-transport-agent>
    citadel-mta
    courier-mta
    esmtp-run
    exim4-daemon-light
    lsb-invalid-mta
    masqmail
    msmtp-mta
    nullmailer
    opensmtpd
    qmail-run
    sendmail-bin
    ssmtp
    xmail
    dma:i386
    dma
    exim4-daemon-heavy
    postfix
 |Recommends: php5-apcu
  Recommends: <php5-apc>
  Conflicts: <owncloud-fhs>
  Conflicts: <owncloud-fhs:i386>
root@fsgo:~#
```

**Comment by RealRancor** [comment]
@enoch85 
See https://github.com/owncloud/core/issues/14188

**Comment by simsala** [comment]
After applying 
sudo apt-get purge php5-apcu
my system has been stable ever since.

**Comment by aproposnix** [comment]
@simsala On Ubuntu 14.04 that didn't work for me. I'm at work right now and my server stopped again despite having purged php5-apcu this morning.

**Comment by osos** [comment]
On Ubuntu server 14.04 i purged php5-apcu and restarted apache2 => worked

I tried disabling apc in htaccess, but with no result.

This is a major problem, as now my other websites does not have apc support.

Did anyone file a bug report to APC/apache, as ownCloud should never be able to crash apache totally.

**Comment by enoch85** [comment]
@osos

> Did anyone file a bug report to APC/apache. 

No, but it would be very nice if you could file a report. :)

**Comment by LukasReschke** [comment]
As discussed with @DeepDiver1975 I'll try to dig into this tomorrow.

**Comment by enoch85** [comment]
@LukasReschke :+1:  Very nice :beers: 

And btw, I purged apc and apcu totally from my system, and now it works since like 36 hours. So I can confirm that this is the issue here.

**Comment by osos** [comment]
Bug reported to launchpad: https://bugs.launchpad.net/ubuntu/+source/php-apcu/+bug/1422484

**Comment by enoch85** [comment]
@osos Thanks alot!

**Comment by enoch85** [comment]
Just for others to know... Downloaded apc 4.0.6 and I'm trying it now. You guys could do the same:

```
wget https://launchpad.net/~ondrej/+archive/ubuntu/php5/+build/6149263/+files/php5-apcu_4.0.6-1%2Bdeb.sury.org~utopic%2B1_amd64.deb
```
```
sudo dpkg -i php5-apcu_4.0.6-1+deb.sury.org~utopic+1_amd64.deb
```

**Comment by Akhenaton** [comment]
Absolutely NO problem here with 

apcu
APCu Support	Enabled
**Version 	4.0.6**
APCu Debugging 	Disabled
MMAP Support 	Enabled
MMAP File Mask 	no value
Serialization Support 	php
Revision 	$Revision: 328290 $
**Build Date Oct 3 2014 08:40:50* *

**Comment by enoch85** [comment]
@Akhenaton Thanks for that confirmation. 

My setup:
Ubuntu 14.04 with PHP 5.5.9. Same Revision as you with the download I posted above. This seems to be the permanent solution.

**Comment by LukasReschke** [comment]
Let me try then adding a version check for APCu…

**Comment by LukasReschke** [comment]
Thanks for all the testing meanwhile. Much appreciated! :rocket:

**Comment by LukasReschke** [comment]
https://github.com/owncloud/core/pull/14273 should "fix" this issue by only activating the APCu cache if at least 4.0.6 is used. – If somebody would be able to review this this would be awesome!

**Comment by enoch85** [comment]
@jnweiger Is there any possibility to include 4.0.6 in the package instead?

**Comment by enoch85** [comment]
**FYI:**

Tested with Zend OPcache and it works as well. Here is a guide for Ubuntu 12.04 and PHP 5.4: https://www.en0ch.se/tips-tricks/install-zend-opcache-ubuntu/

Don't forget to purge APCu first. 

`sudo apt-get purge php5-apcu`

**Comment by wesdonohue** [comment]
It looks like the apt-get purge php5-apcu worked for me

**Comment by cyril23** [comment]
Thanks, fixed the problem for me, too. 
System: Ubuntu 14.04.2 LTS; PHP 5.5.9-1ubuntu4.6 (cli) (built: Feb 13 2015 19:17:11) Zend Engine v2.5.0 with Zend OPcache v7.0.3
Before: php5-apcu 4.0.2-2build1

wget http://de.archive.ubuntu.com/ubuntu/pool/universe/p/php-apcu/php5-apcu_4.0.6-1_amd64.deb
sudo apt-get purge php5-apcu
sudo dpkg -i php5-apcu_4.0.6-1_amd64.deb
sudo service apache2 restart

**Comment by petbadger** [comment]
Just wanted to add that I had this problem too.  I was always having to restart Apache.   The fix in cyril23's comments worked for me.

**Comment by RealRancor** [comment]
@petbadger 
Are you using OC 8.0.2? Normally this version should disable the broken APCu PHP module: https://github.com/owncloud/core/blob/v8.0.2/settings/admin.php#L87-L89

**Comment by simsala** [comment]
I also continued to have stability issues with apcu enabled and 8.0.2.
My impression is that 8.0.2 does NOT resolve the issue.

**Comment by RealRancor** [comment]
ownCloud can't fix an issue in APCu, it just can workaround problems by disabling APCu.

Could be useful to post which ACPu version you're using.

**Comment by simsala** [comment]
Using the version currently installed by Ubuntu 14.4:
apt-cache show php5-apcu
Package: php5-apcu
Priority: optional
Section: universe/php
Installed-Size: 306
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Debian PHP PECL Maintainers <pkg-php-pecl@lists.alioth.debian.org>
Architecture: amd64
Source: php-apcu
Version: 4.0.2-2build1
Replaces: php-apc (<< 4.0.0), php5-user-cache
Provides: php-apc, php5-user-cache
Depends: libc6 (>= 2.14), phpapi-20121212
Suggests: php5-gd
Conflicts: php-apc (<< 4.0.0), php5-user-cache, php5-xcache, php5-yac
Filename: pool/universe/p/php-apcu/php5-apcu_4.0.2-2build1_amd64.deb
Size: 73156

**Comment by jnweiger** [comment]
owncloud-server package now has

    Suggests: php5-apcu (>= 4.0.6) | php-apc

Minimum version number added.

**Comment by petbadger** [comment]
@RealRancor 
I am using owncloud version 8 (I assume version 8.0). I had upgraded as soon as version 8 stable was announced.  

I can't recall what version of php5-apcu I was using.  I did a search of the Ubuntu software catalogue and php5-apcu does not even show up.  Perhaps this means it was installed by something else, like owncloud?  I believe the owncloud version I was using was 7.0.3.

Whatever version I had was purged and replaced with php5-apcu_4.0.6-1_amd64.  

Everything has been working ever since.
